docs/debugging-transactions.md

```diff
@@ -0,0 +1,312 @@
+# Debugging Transactions
+
+## Table of Contents
+1. [Overview](#overview)
+2. [Built-in Remix Debugger](#1-built-in-remix-debugger)
+   - [Steps to Use the Built-in Debugger](#steps-to-use-the-built-in-debugger)
+   - [Example of Debugging a Transaction](#example-of-debugging-a-transaction)
+3. [Advanced Debugging with `debug_traceTransaction`](#2-advanced-debugging-with-debug_traceTransaction)
+   - [Manually Calling `debug_traceTransaction`](#manually-calling-debug_traceTransaction)
+   - [Examples of Using `debug_traceTransaction`](#examples-of-using-debug_tracetransaction)
+      - [With `callTracer`](#1-with-callTracer)
+      - [With `opcodeLogger`](#2-with-opcodeLogger)
+
+## Overview
+
+[Remix IDE](https://remix-project.org/) provides powerful debugging tools that allow developers to inspect and analyze the execution of smart contracts.
```

Comment:
Do we want to make this Remix specific? There are other tools that supports debugging through the `debug_traceTransaction` method, _e.g._, Truffle (although deprecated). Maybe there others. Should we make this document generic and use Remix as a running example?

Reply:
Hmm, I have given Remix’s built-in debugger as an example because its the most used and established tool currently.

I can try to change the javascript examples with the manual calls to our relay to be more generic and runnable outside of Remix

Reply:
Should I also add Truffle examples?

Reply:
> Should I also add Truffle examples?

If it's not much trouble, I'd say yes.

I believe the main point to highlight is that we support any tool that uses `debug_traceTransaction`, Remix being one of the them.

Reply:
Done, added some Truffle docs as well!